pkg/dev/podmandev/pod_test.go

```diff
@@ -221,15 +221,15 @@ func Test_createPodFromComponent(t *testing.T) {
 					Name:          "http",
 					ContainerPort: 8080,
 					Protocol:      "TCP",
-					HostPort:      39001,
+					HostPort:      40001,
```

Comment:
Wondering if checking the values this way might not be error-prone if the port is already taken locally. I had an unrelated local component running in Dev mode (and listening on port 40001), and this test was no longer passing because a different port was picked by `createPodFromComponent`.

Maybe we could make sure to explicitly pass a non-nil slice of ports as `usedPorts`  when calling `createPodFromComponent`, so that a port can be picked from this list.

Or, better to me, we should ideally check that the returned port is within the expected range and that the values for `LocalPort` (in `api.ForwardedPort`) and the `HostPort` (in the Pod Container Spec) are the same.

Reply:
I would prefer the solution to pass ports as usedPorts, so we don't have to rely on the system